src/gsy_e/models/strategy/energy_parameters/heatpump/heat_pump.py

```diff
@@ -231,7 +244,11 @@ def update_cop(
     ):
         """Update the COP of the heat pump in its state class."""
         heat_demand_kJ = self._hp_state.get_heat_demand_kJ(last_time_slot)
-        cop = self._calc_cop(heat_demand_kJ, source_temp_C, last_time_slot)
+        if heat_demand_kJ < FLOATING_POINT_TOLERANCE:
+            cop = self._hp_state.get_cop(last_time_slot)
+        else:
+            cop = self._calc_cop(heat_demand_kJ, source_temp_C, last_time_slot)
```

Comment:
I am second-guessing the initial implementation unfortunately, but shouldn't the COP depend on the produced heat energy and not just the heat demand of the home ?

Reply:
`_calc_cop` calls either the `UniversalCOPModel` or the `IndividualCOPModel` (the latter was provided by Tekniker and uses the heat demand as input. The default model `UniversalCOPModel` ignores this parameter. 

My change here is only to populate the COP value in the hp_state even if the `heat_demand_kJ` is 0. This is true for some of the market slots unfortunately. 
I hope it makes sense.

Reply:
I agree, that we pass the traded energy to this method that makes it independent on the heat_demand=0 case.
Still, there are cases when there is no traded energy (e.g. when there is no cheep energy available and the demand is also 0).
In this case we need to set the COP value from the last market slot in order to have a continues profile for COP. 
Does it make sense?